File: project_euler/Problem091/Problem091.c
#include <stdio.h>

static int common_divisor(int i, int j);
static int minimum(int i, int j);

void Problem091(void) {
    int target = 50;
    int result = target * target * 3;
    for (int i = 1; i < target; ++i) {
        for (int j = 1; j <= target; ++j) {
            int divisor = common_divisor(i, j);
            result += minimum((target - i) * divisor / j, j * divisor / i) << 1;
        }
    }
    printf("%d\n", result);
}

static int common_divisor(int i, int j) {
    while (i % j) {
        int auxiliary = i % j;
        i = j;
        j = auxiliary;
    }
    return j;
}

static int minimum(int i, int j) {
    return (i < j) ? i : j;
}

File: project_euler/Problem091/Problem091.js
function Problem091() {
    let target = 50
    let result = target * target * 3
    for (let i = 1; i < target; ++i) {
        for (let j = 1; j <= target; ++j) {
            let divisor = commonDivisor(i, j)
            result += Math.min(Math.floor((target - i) * divisor / j), Math.floor(j * divisor / i)) << 1
        }
    }
    console.log(result)
}

function commonDivisor(i, j) {
    while (i % j) {
        let auxiliary = i % j
        i = j
        j = auxiliary
    }
    return j
}

module.exports = {
    Problem091 : Problem091
};
